MGS2 Substance SLUS_20554 WS adjustments

MackdannyTwitch opened the pull request:
Metal Gear Solid 2: Substance's Widescreen patch includes code for the Skateboarding demo, however as that is a separate ELF from the rest of the game modes, the widescreen is no longer recognized when launching the skateboarding.

MackdannyTwitch commented:
Tested, ready to go.

CookiePLMonster reviewed (changes requested):
I think it's fine to just move those lines without commenting them out, otherwise LGTM.

CookiePLMonster commented:
Just delete, imo no need to keep them commented out or not.

MackdannyTwitch commented:
Resolved.

CookiePLMonster reviewed (approved):
I suspect that the conditional checks are now redundant in both cheats, but this is something that would need to be tested. I'm fine merging it as-is if you're not comfortable verifying this.

MackdannyTwitch commented:
Conditional checks?

Mrlinkwii commented:
whats the status of this ?

MackdannyTwitch commented:
> Any comment on this?
> 
> Metal Gear Solid 2: Substance * NTSC-U * SLUS-20554 * C539049D
> 
> [60 fps] author=flcl8193 description=60 fps for Cutscenes patch=1,EE,001914F4,word,00000000
> 
> https://forums.pcsx2.net/Thread-60-fps-codes?pid=625168#pid625168

I tested it. It seems fine, so i added it.

MackdannyTwitch commented:
> whats the status of this ?

Everything's fine with it, as far as i can tell... however, I'm still not sure what @CookiePLMonster means by confitional checks. Idk what I'd need to be on the lookout for?

CookiePLMonster commented:
> > whats the status of this ?
> 
> Everything's fine with it, as far as i can tell... however, I'm still not sure what @CookiePLMonster means by confitional checks. Idk what I'd need to be on the lookout for?

The current split patches use E-codes to check for something and apply the patches conditionally Are they not the ELF checks that are redundant now that the cheat is split across ELFs?

MackdannyTwitch commented:
> > > whats the status of this ?
> > 
> > 
> > Everything's fine with it, as far as i can tell... however, I'm still not sure what @CookiePLMonster means by confitional checks. Idk what I'd need to be on the lookout for?
> 
> The current split patches use E-codes to check for something and apply the patches conditionally Are they not the ELF checks that are redundant now that the cheat is split across ELFs?

Beats me. Thats how it was written originally before i moved the Skateboard WS cheat to its own file. When I moved it, I did so because the emulator was making it clear that "no cheats/patches were found" when starting the Skateboarding demo, and the aspectbratio had shifted back to 4:3. NOW, the WS functions on the Skateboarding as intended. Any ideas on how to proceed?

MackdannyTwitch commented:
@CookiePLMonster okay, i actually found the original discussion about adding the conditionals in a 2021 post on the forum. I have tested with the e codes removed and everything still works as it did, so I have now removed them.
